codice/ddf
# [2.11.x] DDF-3335 Added delayed start to Platform Command Scheduler jobs

#### What does this PR do?
This PR fixes an issue where platform-scheduler fails on the first attempt(s) to run command jobs after a restart. There is now a 1 minute delay before `Platform Command Scheduler` jobs are started.
#### Who is reviewing it? 
@peterhuffer @AzGoalie @oconnormi 
#### Choose 2 committers to review/merge the PR. 
@clockard
@lessarderic
#### How should this be tested?

1. Before starting up for the first time, add a `ddf.platform.scheduler.Command` configuration file to `/etc`. When I tested, I used

    _`ddf.platform.scheduler.Command-removeExpiredRecords.config`_
    ```
    command="removeall\ -f\ --expired"
    intervalString="10"
    intervalType="secondInterval"
    
    ```

    I made the interval small so I could see it execute multiple times. I also used a command that will only work after installation to make sure that everything works as expected.

2. Start up DDF.
3. Search for `platform-scheduler` in the ddf log, and confirm that there are no `Platform Command Scheduler` warnings.
4. Complete the installer.
5. Follow the Documentation to schedule a new job, and confirm that the new job starts and works as expected.
6. Restart DDF.
7. Check the logs during start-up, search for `platform-scheduler` in the ddf log, and again confirm that there are `Platform Command Scheduler` warnings.
8. Confirm that the previously-configured jobs are working.

#### Any background context you want to provide?
During start-up, the `Platform Command Scheduler` would attempt to execute commands before required bundles has been started, causing nasty error messages to be logged.
#### What are the relevant tickets?
[DDF-3335](https://codice.atlassian.net/browse/DDF-3335)
#### Checklist:
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
